packages/devtools_app/lib/src/shared/history_viewport.dart

```diff
@@ -106,25 +115,44 @@ class _HistoryViewportState<T> extends State<HistoryViewport<T>> {
               ],
               Expanded(
                 child: widget.onTitleTap == null
-                    ? Text(
-                        title,
-                        style: defaultTitleStyle,
+                    ? Row(
+                        children: [
+                          if (widget.titleIcon != null) Icon(widget.titleIcon),
+                          Text(
+                            title,
+                            style: defaultTitleStyle,
+                          ),
+                        ],
                       )
                     : MouseRegion(
                         cursor: SystemMouseCursors.click,
-                        onExit: (_) => _updateTitleStyle(defaultTitleStyle),
+                        onExit: (_) {
+                          _updateTitleStyle(defaultTitleStyle);
+                          _updateIconColor(defaultTitleStyle.color);
+                        },
                         onEnter: (_) {
                           _updateTitleStyle(
                             defaultTitleStyle.copyWith(
                               color: theme.colorScheme.devtoolsLink,
                             ),
                           );
+                          _updateIconColor(theme.colorScheme.devtoolsLink);
                         },
                         child: GestureDetector(
                           onTap: widget.onTitleTap,
-                          child: Text(
-                            title,
-                            style: _titleStyle ?? theme.textTheme.titleSmall,
+                          child: Row(
+                            children: [
+                              if (widget.titleIcon != null)
+                                Icon(
+                                  widget.titleIcon,
+                                  color: _iconColor,
+                                ),
+                              Text(
```

Comment:
Can you put this in an Expanded? It will probably overflow if the screen is narrow and the file name is long

Reply:
Good catch, done!